src/qibolab/instruments/qblox/sequencer.py

```diff
@@ -49,6 +50,7 @@ def add_waveforms(
             NotEnoughMemory: If the memory needed to store the waveforms in more than the memory avalible.
         """
         pulse_copy = pulse.copy()
+        #
         for sweeper in sweepers:
             if sweeper.pulses and sweeper.parameter == Parameter.amplitude:
                 if pulse in sweeper.pulses:
```

Comment:
This, and the next `for sweeper in sweepers` are a repeated pattern, at least in Qblox, cf.
https://github.com/qiboteam/qibolab/blob/639c23b8baeff6cae2e7341e5f1069f631e11552/src/qibolab/instruments/qblox/cluster_qcm_bb.py#L574-L575
https://github.com/qiboteam/qibolab/blob/639c23b8baeff6cae2e7341e5f1069f631e11552/src/qibolab/instruments/qblox/cluster_qcm_bb.py#L582-L584
https://github.com/qiboteam/qibolab/blob/639c23b8baeff6cae2e7341e5f1069f631e11552/src/qibolab/instruments/qblox/controller.py#L274-L278

But at least in Zurich as well something similar is done:
https://github.com/qiboteam/qibolab/blob/639c23b8baeff6cae2e7341e5f1069f631e11552/src/qibolab/instruments/zhinst.py#L650-L651

We could create a container for sweepers, e.g. a `Sweepers` class, that is exposing at least part of the features being implemented here. It could integrate with the implementation of the `sweeper_recursion`, but this should be investigated in more details.

E.g., typical tasks are:
- checking if a given quantity is being swept, that could be `Parameter.amplitude in sweepers`
- which pulse is swept for a certain quantity, `sweepers[Parameter.amplitude]` (potentially returning a list of pulses)
- getting the values swept `sweeper.values(Parameter.duration)` (a nested list could be returned, since multiple occurrences might happen, even though below is assumed that a single one is present)